src/sinks/loki/sink.rs

```diff
@@ -239,7 +240,9 @@ impl EventEncoder {
             for template in self.labels.values() {
                 if let Some(fields) = template.get_fields() {
                     for field in fields {
-                        event.as_mut_log().remove(field.as_str());
+                        if let Ok(path) = parse_target_path(field.as_str()) {
```

Comment:
This is a bit cheating, since we are still basically doing the `remove` on a string. I'm fine with it for now though.
`Templates` weren't really designed to return paths. We've talked before about having template run VRL code, and this VRL feature could potentially help out here: https://github.com/vectordotdev/vrl/issues/25